packages/react-debug-tools/src/__tests__/ReactHooksInspectionIntegration-test.js

```diff
@@ -268,6 +268,97 @@ describe('ReactHooksInspectionIntegration', () => {
     ]);
   });
 
+  it('should inspect useref without override ref', () => {
+    let current = null;
+    let prevCurrent = null;
+    function Foo(props) {
+      const updateCountRef = React.useRef(0);
+      updateCountRef.current++;
+
+      const [state, setState] = React.useState('A');
+
+      const cbRef = React.useRef();
+      cbRef.current = () => {
+        setState(c => String.fromCharCode(c.charCodeAt(0) + 1));
+      };
+      prevCurrent = current;
+      current = cbRef.current;
+      const handler = () => cbRef.current();
+
+      return (
+        <div className="App">
+          <button onClick={handler}>
+            Component updates {updateCountRef.current} times
+          </button>
+          <p>Current state value: {state}</p>
+        </div>
```

Comment:
I think this test case was copied from the GitHub issue, but it could be trimmed down here to better focus on the thing that's being tested.

I'm going to take a pass at rewriting it to include an added DEV warning. Hold tight.